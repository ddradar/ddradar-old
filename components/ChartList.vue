<template>
  <b-table
    :data="charts"
    striped
    hoverable
    :loading="loading"
    focusable
    :mobile-cards="false"
  >
    <template slot-scope="props">
      <b-table-column field="songName" label="Name">
        <nuxt-link
          :to="
            `/song/${props.row.songId}/${props.row.playStyle}${props.row.difficulty}`
          "
        >
          {{ props.row.songName }}
        </nuxt-link>
      </b-table-column>
      <b-table-column field="playStyle" label="SP/DP" centered>
        {{ getPlayStyleName(props.row.playStyle) }}
      </b-table-column>
      <b-table-column field="difficulty" label="Dif" centered>
        <span class="tag" :class="getDifficutyClassName(props.row.difficulty)">
          {{ getDifficutyName(props.row.difficulty) }}
        </span>
      </b-table-column>
      <b-table-column field="level" label="Lv" numeric>
        {{ props.row.level }}
      </b-table-column>
      <b-table-column field="notes" label="Notes" numeric>
        {{ props.row.notes }}
      </b-table-column>
      <b-table-column field="freezeArrow" label="FA" numeric>
        {{ props.row.freezeArrow }}
      </b-table-column>
      <b-table-column field="shockArrow" label="SA" numeric>
        {{ props.row.shockArrow }}
      </b-table-column>
    </template>

    <template slot="empty">
      <section class="section">
        <div class="content has-text-grey has-text-centered">
          <p>Nothing here.</p>
        </div>
      </section>
    </template>
  </b-table>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { GetDifficultyName } from '@/types/difficulty.enum'
import { GetPlayStyleName } from '@/types/play-style.enum'
import { StepChart } from '@/types/step-chart'

@Component
export default class ChartList extends Vue {
  @Prop() charts!: StepChart[]
  @Prop(Boolean) loading!: boolean

  getPlayStyleName(playStyle: number) {
    return GetPlayStyleName(playStyle)
  }
  getDifficutyName(difficulty: number) {
    return GetDifficultyName(difficulty)
  }
  getDifficutyClassName(difficulty: number) {
    return 'is-' + GetDifficultyName(difficulty).toLowerCase()
  }
}
</script>
