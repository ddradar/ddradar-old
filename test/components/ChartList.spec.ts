import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils'
import Buefy from 'buefy'

import ChartList from '@/components/ChartList.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('ChartList', () => {
  let wrapper: Wrapper<ChartList>
  let vm: any
  beforeEach(() => {
    wrapper = shallowMount(ChartList, { localVue })
    vm = wrapper.vm
  })
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  describe('getPlayStyleName()', () => {
    test('returns "SP" if param is 1', () => {
      expect(vm.getPlayStyleName(1)).toBe('SP')
    })
    test('returns "DP" if param is 2', () => {
      expect(vm.getPlayStyleName(2)).toBe('DP')
    })
    test.each([-1, 0, 1.5, NaN, Infinity, -Infinity])(
      'returns "?" if param is other',
      (i) => {
        expect(vm.getPlayStyleName(i)).toBe('?')
      }
    )
  })
  describe('getDifficultyName()', () => {
    test('returns "BEGINNER" if param is 0', () => {
      expect(vm.getDifficultyName(0)).toBe('BEGINNER')
    })
    test('returns "BASIC" if param is 1', () => {
      expect(vm.getDifficultyName(1)).toBe('BASIC')
    })
    test('returns "DIFFICULT" if param is 2', () => {
      expect(vm.getDifficultyName(2)).toBe('DIFFICULT')
    })
    test('returns "EXPERT" if param is 3', () => {
      expect(vm.getDifficultyName(3)).toBe('EXPERT')
    })
    test('returns "CHALLENGE" if param is 4', () => {
      expect(vm.getDifficultyName(4)).toBe('CHALLENGE')
    })
    test.each([-1, 5, 1.5, NaN, Infinity, -Infinity])(
      'returns "Unknown" if param is other',
      (i) => {
        expect(vm.getDifficultyName(i)).toBe('Unknown')
      }
    )
  })
  describe('getDifficutyClassName()', () => {
    test('returns "is-beginner" if param is 0', () => {
      expect(vm.getDifficultyClassName(0)).toBe('is-beginner')
    })
    test('returns "is-basic" if param is 1', () => {
      expect(vm.getDifficultyClassName(1)).toBe('is-basic')
    })
    test('returns "is-difficult" if param is 2', () => {
      expect(vm.getDifficultyClassName(2)).toBe('is-difficult')
    })
    test('returns "is-expert" if param is 3', () => {
      expect(vm.getDifficultyClassName(3)).toBe('is-expert')
    })
    test('returns "is-challenge" if param is 4', () => {
      expect(vm.getDifficultyClassName(4)).toBe('is-challenge')
    })
    test.each([-1, 5, 1.5, NaN, Infinity, -Infinity])(
      'returns "is-unknown" if param is other',
      (i) => {
        expect(vm.getDifficultyClassName(i)).toBe('is-unknown')
      }
    )
  })
})
