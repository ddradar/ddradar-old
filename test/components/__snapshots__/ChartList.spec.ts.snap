// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartList renders correctly 1`] = `
<div
  class="b-table"
>
  <!---->
   
  <!---->
   
  <div
    class="table-wrapper"
  >
    <table
      class="table is-striped is-hoverable"
      tabindex="0"
    >
      <!---->
       
      <tbody>
        <tr
          class=""
          draggable="false"
        >
          <!---->
           
          <!---->
           
          <td
            class=""
            data-label="Name"
          >
            <a>
              
        AM-3P
      
            </a>
          </td>
           
          <td
            class="has-text-centered"
            data-label="SP/DP"
          >
            
      SP
    
          </td>
           
          <td
            class="has-text-centered"
            data-label="Dif"
          >
            <span
              class="tag is-beginner"
            >
              
        BEGINNER
      
            </span>
          </td>
           
          <td
            class="has-text-right"
            data-label="Lv"
          >
            
      3
    
          </td>
           
          <td
            class="has-text-right"
            data-label="Notes"
          >
            
      74
    
          </td>
           
          <td
            class="has-text-right"
            data-label="FA"
          >
            
      0
    
          </td>
           
          <td
            class="has-text-right"
            data-label="SA"
          >
            
      0
    
          </td>
           
          <!---->
        </tr>
         
        <!---->
         
        <!---->
        <tr
          class=""
          draggable="false"
        >
          <!---->
           
          <!---->
           
          <td
            class=""
            data-label="Name"
          >
            <a>
              
        AM-3P
      
            </a>
          </td>
           
          <td
            class="has-text-centered"
            data-label="SP/DP"
          >
            
      SP
    
          </td>
           
          <td
            class="has-text-centered"
            data-label="Dif"
          >
            <span
              class="tag is-basic"
            >
              
        BASIC
      
            </span>
          </td>
           
          <td
            class="has-text-right"
            data-label="Lv"
          >
            
      7
    
          </td>
           
          <td
            class="has-text-right"
            data-label="Notes"
          >
            
      164
    
          </td>
           
          <td
            class="has-text-right"
            data-label="FA"
          >
            
      0
    
          </td>
           
          <td
            class="has-text-right"
            data-label="SA"
          >
            
      0
    
          </td>
           
          <!---->
        </tr>
         
        <!---->
         
        <!---->
        <tr
          class=""
          draggable="false"
        >
          <!---->
           
          <!---->
           
          <td
            class=""
            data-label="Name"
          >
            <a>
              
        AM-3P
      
            </a>
          </td>
           
          <td
            class="has-text-centered"
            data-label="SP/DP"
          >
            
      SP
    
          </td>
           
          <td
            class="has-text-centered"
            data-label="Dif"
          >
            <span
              class="tag is-difficult"
            >
              
        DIFFICULT
      
            </span>
          </td>
           
          <td
            class="has-text-right"
            data-label="Lv"
          >
            
      8
    
          </td>
           
          <td
            class="has-text-right"
            data-label="Notes"
          >
            
      195
    
          </td>
           
          <td
            class="has-text-right"
            data-label="FA"
          >
            
      0
    
          </td>
           
          <td
            class="has-text-right"
            data-label="SA"
          >
            
      0
    
          </td>
           
          <!---->
        </tr>
         
        <!---->
         
        <!---->
      </tbody>
       
      <!---->
    </table>
  </div>
   
  <!---->
</div>
`;

exports[`ChartList renders loading 1`] = `
<div
  class="b-table is-loading"
>
  <!---->
   
  <!---->
   
  <div
    class="table-wrapper"
  >
    <table
      class="table is-striped"
      tabindex="0"
    >
      <!---->
       
      <tbody>
        <tr
          class="is-empty"
        >
          <td
            colspan="0"
          >
            <section
              class="section"
            >
              <div
                class="content has-text-grey has-text-centered"
              >
                <p>
                  Nothing here.
                </p>
              </div>
            </section>
          </td>
        </tr>
      </tbody>
       
      <!---->
    </table>
  </div>
   
  <!---->
</div>
`;
